### DISCUSSION TITLE
appimage magick cannot open jxl file

### DISCUSSION THREAD

**Discussion by braindevices** [ImageMagick/ImageMagick#6676] [Contribute]
### ImageMagick version

7.1.0-47 Q16-HDRI x86_64 d91623c12:20220827

### Operating system

Linux

### Operating system, version and so on

Ubuntu20.04

### Description

cannot read jxl

```
Delegates (built-in): bzlib djvu fontconfig freetype jbig jng jpeg lcms lqr lzma openexr png raqm tiff webp x xml zlib
```
error:
```
identify: no decode delegate for this image format `JXL' @ error/constitute.c/ReadImage/741.
```

### Steps to Reproduce

`magick identify any.jxl`

**Comment by snibgo** [comment]
IM can successfully read JXL files when it has a JXL built-in delegate.

**Comment by fmw42** [comment]
I suspect that the Imagemagick AppImage is not going to be configured for every possible image format. That might be too much bloat. I think it is configured for the most frequently used image formats.

**Comment by braindevices** [comment]
well, build the IM is quite a job since there are lots of dependencies. I would expect the appimage provide the full functionality. If it is too bloat (I doubt the additional Delegates will make to much bigger though), can we make it 2 versions, one only contains the common format, the other contains all formats?

**Comment by braindevices** [comment]
and JPEG XL is more and more common right now

**Comment by fmw42** [comment]
The IM developers do not know what formats every one needs.  They will be different for everyone.  

Look at the list of all formats by

`convert -list format`

That is over 250 formats.  That would be quite some bloat!

**Reply by carlosvsilva** [reply]
Instagram accepts HDR JXLs, Adobe released support to export JXL in Lightroom, and major smartphone vendors enabled camera JXL output recently, only Chrome is one of the last famous temporary holdouts regarding JXL support, so I vote on promoting JXL support to the default AppImage :) Cheers
